hooks/pre_build

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+if [[ "$SOURCE_BRANCH" = "develop" ]]; then
+    # add -d flag to checkout.bash and update image dependencies
+    sed -i "s|checkout.bash|checkout.bash -d|g; s|usdotfhwastol/|usdotfhwastoldev/|g; s|:[0-9]*\.[0-9]*\.[0-9]*|:develop|g s|:CARMA[a-zA-Z]*_[0-9]*\.[0-9]*\.[0-9]*|:develop|g; s|:carma-[a-zA-Z]*-[0-9]*\.[0-9]*\.[0-9]*|:develop|g" \
+        Dockerfile
+elif [[ "$SOURCE_BRANCH" =~ ^release/.*$ ]]; then
+    # swap checkout branch in checkout.bash to release branch
+    sed -i "s|usdotfhwastol/|usdotfhwastolcandidate/|g; s|:[0-9]*\.[0-9]*\.[0-9]*|:candidate|g; s|:CARMA[a-zA-Z]*_[0-9]*\.[0-9]*\.[0-9]*|:candidate|g; s|:carma-[a-zA-Z]*-[0-9]*\.[0-9]*\.[0-9]*|:candidate|g" \
```

Comment:
Not sure if this branch is called for the release build, but I'm not sure it would respect usdotfhwastoldev if that is present in the Dockerfile.

Reply:
Yeah this is for candidate builds. i'll add a usdotfhwastoldev/ to the command.